docker/Dockerfile

```diff
@@ -0,0 +1,445 @@
+# USAGE INSTSRUCTIONS
+# 
+# Build image (run this in the main directory (where .git folder is)):
+# docker build -t precice/fenicsx -f ./docker/Dockerfile .
+#
+# Run image:
+# docker run -it --rm precice/fenicsx
+#
+# Run image and map current directory to home/workdir on Linux:
+# docker run -it --rm -v $PWD:/home/workdir precice/fenicsx
+# and on Windows:
+# docker run -it --rm -v ${PWD}:/home/workdir precice/fenicsx
+
+FROM ubuntu:21.10
```

Comment:
@arvedes Can you try to use 
```
# Dockerfile to build a ubuntu image containing the installed Debian package of a release 
ARG branch=develop
ARG from=precice/python-bindings:${branch}
FROM $from
```
here? This would be more consistent with [the Dockerfile of the fenics-adapter](https://github.com/precice/fenics-adapter/blob/develop/tools/releasing/packaging/docker/Dockerfile) and saves us the precice installation at the bottom.

If this leads to tricky problems with dependencies: Let's keep it as it is. I guess there will be more convenient ways for installing fenics-x in the future and for development this Dockerfile should already help a lot.

Reply:
I'm understanding #6 to keep the docker file as it is. We can still update it at a later point in time. I will resolve it here.